google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/util/AppEngineUtil.java

```diff
@@ -16,34 +16,64 @@
 
 package com.google.cloud.tools.intellij.appengine.util;
 
-import org.joda.time.DateTime;
-import org.joda.time.format.DateTimeFormat;
-import org.joda.time.format.DateTimeFormatter;
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import com.google.gson.reflect.TypeToken;
+
+import java.lang.reflect.Type;
+import java.util.List;
 
 /**
  * App Engine utility methods
  */
 public class AppEngineUtil {
 
+  // Just to return two strings.
+  public static class VersionService {
+    public String version = null;
+    public String service = null;
+  }
+
+  // Holds de-serialized JSON output of gcloud app deploy.
+  private static class DeployOutput {
+    private static class versionElement {
```

Comment:
I don't know if it makes sense for it to be static since its related to an instance of the deployment. I think it would be fine to put your updated gson logic back into a private method of the DeployAction class.